compositor_render/src/scene/view_component.rs

```diff
@@ -1,11 +1,25 @@
-use compositor_common::{scene::Resolution, util::colors::RGBAColor};
+use compositor_common::scene::Resolution;
 
 use crate::{
     scene::ViewChildrenDirection,
     transformations::layout::{Layout, LayoutContent, NestedLayout},
 };
 
-use super::{components::ViewComponent, BaseNode, BuildSceneError, Component, LayoutComponent};
+use super::{
+    components::ViewComponent, BaseNode, BuildSceneError, Component, LayoutComponent, Position,
+};
+
+#[derive(Debug)]
+struct StaticChildLayoutOpts {
+    width: Option<usize>,
+    height: Option<usize>,
+    /// offset inside parent component
+    static_offset: usize,
+    /// For direction=row defines width of a dynamically sized component
+    /// For direction=column defines height of a dynamically sized component
+    dynamic_child_size: usize,
+    parent_size: Resolution,
+}
 
 impl ViewComponent {
     pub(super) fn base_node(&self) -> Result<BaseNode, BuildSceneError> {
```

Comment:
I forgot to add that in the previous review:
The name `BaseNode` threw me off quite a bit. It confused me for an embarrassingly long time when I was looking over the last PR. I think the name should be changed, but I still don't fully understand what this is so I can't really think of a better name.

Reply:
Did you see the comment on the BaseNode type?
```
/// Intermediate representation of a node tree while it's being constructed.
```

> but I still don't fully understand what this is so I can't really think of a better name.

BaseNode is an intermidiate representation. When we are creating a Node tree from the Component tree we go through the component tree recursively merging components into nodes if possible. We need that intermediate type to add special logic for layout components that are node roots.